src/plugins/intel_gpu/src/runtime/execution_config.cpp

```diff
@@ -235,6 +236,15 @@ void ExecutionConfig::apply_user_properties(const cldnn::device_info& info) {
     if (info.supports_immad) {
         set_property(ov::intel_gpu::queue_type(QueueTypes::in_order));
     }
+    if (!is_set_by_user(ov::hint::enable_cpu_reservation)) {
+        bool enable_cpu_pinning = get_property(ov::hint::enable_cpu_pinning);
+        if (enable_cpu_pinning) {
+            set_property(ov::hint::enable_cpu_reservation(true));
+        }
+    }
+    if (get_property(ov::hint::enable_cpu_reservation) && !get_property(ov::hint::enable_cpu_pinning)) {
+        set_property(ov::hint::enable_cpu_pinning(true));
```

Comment:
Why we need this logic here? User can set enable_cpu_reservation to true and enable_cpu_pinning to false.

Reply:
I think if enable_cpu_reservation is true, enable_cpu_pinning must be true. Because reserving cpus means pinning   tasks to fixed cpus firstly, then reserving these cpus.

Reply:
User still can use enable_cpu_reservation true with enable_cpu_pinning false together. Then the next model only can use remained CPU cores for latency/throughput hit.

Reply:
Ok, done.